app/lib/katello/concerns/base_template_scope_extensions.rb

```diff
@@ -117,7 +117,7 @@ def load_errata_applications(filter_errata_type: nil, include_last_reboot: 'yes'
                               select: 'foreman_tasks_tasks.*,template_invocations.id AS template_invocation_id',
                               search: search
         )
-        only_host_ids = ::Host.search_for(host_filter).pluck(:id) if host_filter
+        only_host_ids = ::Host.search_for(host_filter).pluck(:id)
```

Comment:
We should fix this instead, without the filter we fetch all the hosts, that might be quite a lot.

Reply:
something along the lines of

```ruby
# line #132
seen_host_ids &= only_host_ids if only_host_ids
```
```ruby
# line #138
next unless only_host_ids && only_host_ids.include?(task.input['host']['id'].to_i)
```

Reply:
Thanks @ezr-ondrej , I have update the PR